src/pages/TradeV3/perps/components/CollateralPanel.tsx

```diff
@@ -1,198 +1,93 @@
+import { TooltipTrigger, TooltipContent, Tooltip, cn } from 'gfx-component-lib'
+import { InfoImage, InfoLabel, PerpsLayout } from './PerpsGenericComp'
+import { Connect } from '../../../../layouts'
 
-const TABS_WRAPPER = styled.div<{ $isLocked: boolean }>`
-  .header-wrapper {
-    display: flex;
-    height: 30px;
-
-    > div {
-      width: 34%;
-      height: 100%;
-    }
-
-    .disable {
-      cursor: not-allowed;
-    }
-  }
-  .tab {
-    border: ${({ theme }) => '1px solid ' + theme.tokenBorder};
-    border-top: none;
-    cursor: pointer;
-  }
-  .active {
-    ${tw`text-[#3C3C3C] dark:text-grey-5`}
-    background: linear-gradient(94deg, #f7931a 0%, #ac1cc7 100%);
-    padding: 2px;
-  }
-  .white-background {
-    background-color: ${({ theme }) => theme.bg2};
-    width: 100%;
-    height: 100%;
-  }
-  .activeTab {
-    background-image: linear-gradient(to right, rgba(247, 147, 26, 0.4) 0%, rgba(172, 28, 199, 0.4) 100%);
-  }
-  .field {
-    ${tw`dark:text-grey-5 text-grey-1 flex justify-center items-center text-tiny font-semibold h-full`}
-  }
-  ${tw`h-full w-full relative overflow-y-hidden`}
-  filter: blur(${({ $wallet }) => ($wallet ? 0 : 3)}px) !important;
-  border: ${({ theme }) => '1px solid ' + theme.tokenBorder};
-`
-
-const WRAPPER = styled.div<{ $height: boolean }>`
-  padding: ${({ $height }) => ($height > 900 ? '20px 15px' : '8px 15px')};
-  .coin-icon {
-    ${tw`mr-2.5`}
-  }
-  .heart-icon {
-    ${tw`ml-2`}
-  }
-  .tooltipIcon {
-    ${tw`!w-3.5 !h-3.5 ml-0 cursor-pointer`}
-  }
-  .health-icon {
-    ${tw`flex items-center`}
-  }
-  .bar-holder {
-    ${tw`flex`}
-  }
-  .bars {
-    ${tw`h-5 w-1.5 mr-2.5 inline-block rounded-average`}
-  }
-  .bars:last-child {
-    ${tw`mr-4`}
-  }
-  .green {
-    ${tw`bg-green-2`}
-  }
-  .red {
-    ${tw`bg-red-1`}
-  }
-  .greenText {
-    ${tw`!text-green-2`}
-  }
-  .redText {
-    ${tw`!text-red-1`}
-  }
-  .yellow {
-    ${tw`bg-[#F0B865]`}
-  }
-  .gray {
-    background: ${({ theme }) => theme.bg24};
-  }
-  .separator {
-    ${tw`mb-3.75 border-b-[5px] border-dashed border-[#4a4a4a]`}
-  }
-`
-
-const ACCOUNT_ROW = styled.div<{ $height: boolean }>`
-  ${tw`flex flex-row justify-between items-start`}
-  margin-bottom: ${({ $height }) => ($height > 900 ? '18px' : '16px')};
-  line-height: normal;
-  .key {
-    ${tw`text-tiny font-semibold text-[#636363] dark:text-[#B5B5B5]`}
-  }
-  .value {
-    ${tw`text-tiny font-semibold text-[#3C3C3C] dark:text-grey-5`}
-  }
-  .balances {
-    ${tw`text-right`}
-    > div {
-      ${tw`text-right mb-[2px]`}
-    }
-  }
-  .positive {
-    ${tw`text-regular font-black text-green-2`}
-  }
-  .negative {
-    ${tw`text-regular font-black text-red-1`}
-  }
-  .tooltip-row {
-    ${tw`flex items-center`}
-  }
-`
+export const CollateralPanel: FC = (): JSX.Element => {
+  const { wallet } = useWallet()
+  const publicKey = useMemo(() => wallet?.adapter?.publicKey, [wallet?.adapter?.publicKey, wallet?.adapter])
```

Comment:
you don't have to do this ever again :D You can use the `connectedWalletPublicKey` from `useWalletBalance` to save some time/effort